ios/brave-ios/Sources/AIChat/Components/Messages/AIChatResponseMessageView.swift

```diff
@@ -45,51 +47,159 @@ struct AIChatResponseMessageView: View {
               .font(.subheadline)
               .foregroundStyle(Color(braveSystemName: .textPrimary))
               .multilineTextAlignment(.leading)
               .frame(maxWidth: .infinity, alignment: .leading)
           }
-        }
-      } else {
-        // Render Plain Text Block
-        Text(prompt)
-          .font(.subheadline)
-          .foregroundStyle(Color(braveSystemName: .textPrimary))
-          .multilineTextAlignment(.leading)
+          .fixedSize()
           .frame(maxWidth: .infinity, alignment: .leading)
+        }
+      }
+
+      if let searchQueriesEvent = searchQueriesEvent, !isEntryInProgress {
+        HStack {
+          Image(braveSystemName: "leo.search")
+
+          renderSearchQueries(queries: searchQueriesEvent.searchQueries)
+            .tint(Color(braveSystemName: .textPrimary))
+            .multilineTextAlignment(.leading)
+            .frame(maxWidth: .infinity, alignment: .leading)
+        }
+        .padding(.top)
       }
     }
     .fixedSize(horizontal: false, vertical: true)
   }
+
+  @ViewBuilder
+  func renderMarkdown(text: String) -> some View {
+    if let textBlocks = MarkdownParser.parse(
+      string: text,
+      preferredFont: .preferredFont(forTextStyle: .subheadline),
+      useHLJS: true,
+      isDarkTheme: true
+    ) {
+      ForEach(textBlocks, id: \.self) { block in
+        // Render Code Block
+        if let codeBlock = block.codeBlock {
+          Text(block.string)
+            .font(.subheadline)
+            .foregroundStyle(Color(braveSystemName: .textPrimary))
+            .multilineTextAlignment(.leading)
+            .frame(maxWidth: .infinity, alignment: .leading)
+            .padding()
+            .background(
+              RoundedRectangle(cornerRadius: 8.0, style: .continuous)
+                .fill(codeBlock.backgroundColor)
+            )
+            .clipShape(RoundedRectangle(cornerRadius: 8.0, style: .continuous))
+        } else {
+          // Render Text Block
+          Text(block.string)
+            .font(.subheadline)
+            .foregroundStyle(Color(braveSystemName: .textPrimary))
+            .multilineTextAlignment(.leading)
+            .frame(maxWidth: .infinity, alignment: .leading)
+        }
+      }
+    } else {
+      // Render Plain Text Block
+      Text(text)
+        .font(.subheadline)
+        .foregroundStyle(Color(braveSystemName: .textPrimary))
+        .multilineTextAlignment(.leading)
+        .frame(maxWidth: .infinity, alignment: .leading)
+    }
+  }
+
+  func renderSearchQueries(queries: [String]) -> Text {
+    let font = Font.callout
+
+    var text = AttributedString(Strings.AIChat.leoImprovedAnswerBraveSearch)
+    text.font = font
+    text.foregroundColor = Color(braveSystemName: .textTertiary)
+
+    var space = AttributedString(" ")
+    space.font = font
+    space.foregroundColor = Color(braveSystemName: .textTertiary)
+
+    text = text + space
```

Comment:
nit: `text` is already mutable, you may as well mutate in-place instead (`text.append(space)` or `text += space`) instead of re-assigning